### PR TITLE
Upcoming & past calendar events

### PR DESCRIPTION
**Pull request by bitfriend**
Will fix https://github.com/acterglobal/a3/issues/967

### CODE REVIEW

File: native/acter/src/api/calendar_events.rs [hunk 1]
```diff
@@ -237,6 +229,26 @@ impl CalendarEvent {
...
+        let client = self.client.clone();
+        let event_id = self.inner.event_id().to_owned();
+        let my_id = self.client.user_id().context("User not found")?;
+
+        RUNTIME
+            .spawn(async move {
+                let manager =
+                    models::RsvpManager::from_store_and_event_id(client.store(), &event_id).await;
+                // read last rsvp
+                for entry in manager.rsvp_entries().await?.into_iter().rev() {
```
Comment (gnunicorn) [inline native/acter/src/api/calendar_events.rs:243]: maybe `rsvp_entries` should actually be a hashmap `userid -> latest status` and then we just overwrite the latest value? Not necessarily something to be changed today though...

File: native/acter/src/api/calendar_events.rs [hunk 2]
```diff
@@ -237,6 +229,26 @@ impl CalendarEvent {
...
+    pub async fn my_rsvp_status_str(&self) -> Result<String> {
+        let client = self.client.clone();
+        let event_id = self.inner.event_id().to_owned();
+        let my_id = self.client.user_id().context("User not found")?;
+
+        RUNTIME
+            .spawn(async move {
+                let manager =
+                    models::RsvpManager::from_store_and_event_id(client.store(), &event_id).await;
+                // read last rsvp
```
Comment (gnunicorn) [inline native/acter/src/api/calendar_events.rs:242]: just saw below that `rsvp_manager.my_status` exists. Why not use that?
```
                        return rsvp_manager.my_status().await?.text();
```

**Review by gnunicorn — APPROVED** [review]
Looking good. Just a minor style question.

File: native/acter/src/api/calendar_events.rs (hunk 1 above)
Comment (bitfriend) [reply native/acter/src/api/calendar_events.rs]: Fixed by https://github.com/acterglobal/a3/pull/979/commits/65e51eac4581093951a14e1ff44c21eba7004ce9

File: native/acter/src/api/calendar_events.rs (hunk 2 above)
Comment (bitfriend) [reply native/acter/src/api/calendar_events.rs]: Fixed by https://github.com/acterglobal/a3/pull/979/commits/f6a9a779b99fbfacd38031c7fbebd2825e9eb2ca

**Review by gnunicorn — APPROVED** [review]

**Comment by gnunicorn** [comment]
@bitfriend your integration test is failing though.

**Pull request merged**